src/platform/forms-system/src/js/components/FormNav.jsx

```diff
@@ -11,82 +10,112 @@ import {
+  const showHeader = Math.abs(current - index) === 1;
+
+  // The goal with this is to quickly "remove" the header from the DOM, and
+  // immediately re-render the component with the header included.
+  // `current` changes when the form chapter changes, and when this happens
+  // we want to force react to remove the <h2> and re-render it. This should
+  // ensure that VoiceOver on iOS will pick up on the new <h2>
+  // https://github.com/department-of-veterans-affairs/va.gov-team/issues/12323
+  useEffect(
+    () => {
+      if (!environment.isProduction()) {
+        if (current > index + 1) {
+          setIndex(index + 1);
+        } else if (current === index) {
+          setIndex(index - 1);
+        }
       }
-    }
 
-    return (
-      <div>
-        <SegmentedProgressBar total={chapters.length} current={current} />
-        <div className="schemaform-chapter-progress">
-          <div
-            aria-valuenow={current}
-            aria-valuemin="1"
-            aria-valuetext={`Step ${current} of ${
-              chapters.length
-            }: ${chapterName}`}
-            aria-valuemax={chapters.length}
-            className="nav-header nav-header-schemaform"
-          >
-            {!environment.isProduction() ? (
-              <h2 className="vads-u-font-size--h4">
-                {`Step ${current} of ${chapters.length}: ${chapterName}`}
-              </h2>
-            ) : (
-              <h2 className="vads-u-font-size--h4">
-                <span className="form-process-step current">{current}</span>{' '}
-                <span className="form-process-total">of {chapters.length}</span>{' '}
-                {chapterName}
+      return () => {
+        if (!environment.isProduction()) {
+          focusElement('.nav-header > h2');
+        }
+      };
+    },
+    [current, index],
+  );
+
+  return (
+    <div>
+      <SegmentedProgressBar total={chapters.length} current={current} />
+      <div className="schemaform-chapter-progress">
+        <div
+          aria-valuenow={current}
+          aria-valuemin="1"
+          aria-valuetext={stepText}
+          aria-valuemax={chapters.length}
+          className="nav-header nav-header-schemaform"
+        >
+          {!environment.isProduction() &&
+            showHeader && (
+              <h2 id="nav-form-header" className="vads-u-font-size--h4">
+                {stepText}
               </h2>
             )}
-          </div>
+          {!environment.isProduction() &&
+            !showHeader && (
+              <div className="vads-u-font-size--h4">{stepText}</div>
+            )}
+
+          {environment.isProduction() && (
+            <h2 className="vads-u-font-size--h4">
+              <span className="form-process-step current">{current}</span>{' '}
+              <span className="form-process-total">of {chapters.length}</span>{' '}
+              {chapterName}
+            </h2>
+          )}
```

Comment:
This is admittedly awkward, but it should be very temporary. Lines 97-106 are effectively a ternary for non-prod environments. This "ternary" will show the actual `<h2>` under normal circumstances, but when we manipulate the `index` state variable to be far enough away from `current`, that means that we don't want to render the `h2` and will render the fake `div` header instead.